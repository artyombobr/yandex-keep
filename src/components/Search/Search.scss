.search {
  display: flex;
  width: 100%;

  &__input{
    font-size: 15px;
    line-height: 32px;
    width: 100%;
    border: 2px solid #E6E6E6;
    box-sizing: border-box;
    border-radius: 4px 0 0 4px;
    padding-left: 12px;
    outline: none;
    transition: border .3s;

    &:focus {
      border: 2px solid #ffdb4d;
    }
  }

  &__span{
    position: relative;

    &::after{
      content: url("svg/reset.svg");
      position: absolute;
      top: 7px;
      right: 11px;
      height: 10px;
      width: 10px;
      cursor: pointer;
    }
  }

  &__button{
    font-size: 15px;
    line-height: 32px;
    border: none;
    outline: none;
    height: 100%;
    padding: 0 16px;
    background: #E6E6E6;
    border-radius: 0 4px 4px 0;
    cursor: pointer;
    transition: background .3s
  }

  &:hover .search__input {
    border: 2px solid #d6d6d6;
  }

  &:hover .search__button {
    background: #d7d7d7;
  }
}
