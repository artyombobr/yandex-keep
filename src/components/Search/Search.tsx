import React, { useEffect, useState } from 'react';
import Fuse from 'fuse.js';
import './Search.scss';
import { connect } from 'react-redux';
import { setVisibleNotes } from '../../actions';

const options = {
  threshold: 0.2,
  minMatchCharLength: 1,
  keys: ['title', 'text', 'attachments.url'],
};

const Search = (props: any) => {
  const { className, allNotes, dispatch } = props;
  const [text, setText] = useState('');
  useEffect(() => {
    const fuse = new Fuse(allNotes, options);
    const result = fuse.search(text);
    if (result.length || text.length) {
      dispatch(setVisibleNotes(result));
    } else {
      dispatch(setVisibleNotes(allNotes));
    }
  }, [allNotes, text]);
  return (
    <form className={`${className} search`}>
      <input
        className="search__input"
        type="text"
        value={text}
        placeholder="Поиск"
        onChange={e => setText(e.target.value)}
      />
      <span className="search__span" onClick={() => setText('')} />
      <button className="search__button" type="button">
        {'Найти'}
      </button>
    </form>
  );
};

const stateToProps = (state: any) => {
  return {
    allNotes: state.allNotes,
  };
};

export default connect(stateToProps)(Search);
