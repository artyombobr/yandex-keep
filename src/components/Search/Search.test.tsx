import React from 'react';
import { shallow } from 'enzyme';
import { Search } from './Search';

describe('Search', () => {
  it('correct render', () => {
    const component = shallow(<Search />);
    expect(component).toMatchSnapshot();
  });
});
