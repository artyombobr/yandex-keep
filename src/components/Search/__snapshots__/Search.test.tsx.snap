// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="search__input"
          onChange={[Function]}
          placeholder="Поиск"
          type="text"
          value=""
        />,
        <span
          className="search__span"
          onClick={[Function]}
        />,
        <button
          className="search__button"
          type="button"
        >
          Найти
        </button>,
      ],
      "className": "undefined search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search__input",
          "onChange": [Function],
          "placeholder": "Поиск",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search__span",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Найти",
          "className": "search__button",
          "type": "button",
        },
        "ref": null,
        "rendered": "Найти",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="search__input"
            onChange={[Function]}
            placeholder="Поиск"
            type="text"
            value=""
          />,
          <span
            className="search__span"
            onClick={[Function]}
          />,
          <button
            className="search__button"
            type="button"
          >
            Найти
          </button>,
        ],
        "className": "undefined search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search__input",
            "onChange": [Function],
            "placeholder": "Поиск",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search__span",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Найти",
            "className": "search__button",
            "type": "button",
          },
          "ref": null,
          "rendered": "Найти",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
